sw/airborne/modules/computer_vision/opticflow_module.c

```diff
@@ -70,15 +81,18 @@ struct image_t *opticflow_module_calc(struct image_t *img);     ///< The main op
 static void opticflow_telem_send(struct transport_tx *trans, struct link_device *dev)
 {
   pthread_mutex_lock(&opticflow_mutex);
-  if (opticflow_result.noise_measurement < 0.8) {
-    pprz_msg_send_OPTIC_FLOW_EST(trans, dev, AC_ID,
-                                 &opticflow_result.fps, &opticflow_result.corner_cnt,
-                                 &opticflow_result.tracked_cnt, &opticflow_result.flow_x,
-                                 &opticflow_result.flow_y, &opticflow_result.flow_der_x,
-                                 &opticflow_result.flow_der_y, &opticflow_result.vel_body.x,
-                                 &opticflow_result.vel_body.y, &opticflow_result.vel_body.z,
-                                 &opticflow_result.div_size, &opticflow_result.surface_roughness,
-                                 &opticflow_result.divergence); // TODO: no noise measurement here...
+  for (int idx_camera = 0; idx_camera < ACTIVE_CAMERAS; idx_camera++) {
+    if (opticflow_result[idx_camera].noise_measurement < 0.8) {
+      pprz_msg_send_OPTIC_FLOW_EST(trans, dev, AC_ID,
+                                   &opticflow_result[idx_camera].fps, &opticflow_result[idx_camera].corner_cnt,
+                                   &opticflow_result[idx_camera].tracked_cnt, &opticflow_result[idx_camera].flow_x,
+                                   &opticflow_result[idx_camera].flow_y, &opticflow_result[idx_camera].flow_der_x,
+                                   &opticflow_result[idx_camera].flow_der_y, &opticflow_result[idx_camera].vel_body.x,
+                                   &opticflow_result[idx_camera].vel_body.y, &opticflow_result[idx_camera].vel_body.z,
+                                   &opticflow_result[idx_camera].div_size,
+                                   &opticflow_result[idx_camera].surface_roughness,
+                                   &opticflow_result[idx_camera].divergence); // TODO: no noise measurement here...
+    }
```

Comment:
Could you expand this message to add the camera id to the telemetry?

Reply:
In order to change the periodic telemetry I added a message.xml and recompiled paparazzi to make sure the generated files were also updated with the new message field. I don't see a messages.xml in master, how can I commit these changes without causing a build failure in semaphoreci?

Reply:
That's correct, there is no messages.xml in master ;). To include your new message:
1. Make a new branch from [pprzlink](https://github.com/paparazzi/pprzlink)
2. Add your new message definition to [message_definitions/v1.0/messages.xml](https://github.com/paparazzi/pprzlink/blob/master/message_definitions/v1.0/messages.xml)
3. Make an additional pull request at the pprzlink repo (and post the PR number here as a reminder ;) )
Once the pprzlink pull request has been merged, you can update the pprzlink submodule in *this* pull request to include your message into master. HTH